### PR TITLE
Collapsed menu hover effect

### PR DESCRIPTION
**Pull request by jomunker**
This PR adds the hover functionality for the collapsed menu.

## Screen recording

https://github.com/vivid-planet/comet/assets/56400587/08701731-46af-4fa8-8e0a-ff8bbeda3360

### CODE REVIEW

**Comment by johnnyomair** [comment]
IMO review requests should only be requested once the PR is ready to review (not WIP:) 🙂

File: demo/admin/src/App.tsx [hunk 1]
```diff
@@ -142,6 +142,7 @@ class App extends React.Component {
                                                                                         path={`${match.path}/project-snips/main-menu`}
                                                                                         component={MainMenu}
                                                                                     />
+                                                                                    <RouteWithErrorBoundary path={`${match.path}/subsubpage`} />
```
Comment (johnnyomair) [inline demo/admin/src/App.tsx:145]: This should be a Dev Story in Storybook instead. Adding a useless route to Demo isn't right

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 2]
```diff
@@ -1,43 +1,28 @@
...
-type MenuChild = React.ReactElement<MenuItemRouterLinkProps>;
+export type MenuChild = React.ReactElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>;
 
 export interface MenuCollapsibleItemProps extends MenuItemProps {
     children: MenuChild | MenuChild[];
     openByDefault?: boolean;
-    openedIcon?: React.ReactNode;
-    closedIcon?: React.ReactNode;
+    OpenedIcon?: React.JSXElementConstructor<SvgIconProps>;
+    ClosedIcon?: React.JSXElementConstructor<SvgIconProps>;
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:23]: Why change this? It is a breaking change. Also, we use lower case prop names for other React components...

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 3]
```diff
@@ -1,43 +1,28 @@
...
+    isMenuOpen?: boolean;
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:24]: It should be possible to get this information from the context

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 4]
```diff
@@ -46,53 +31,160 @@ const CollapsibleItem: React.FC<WithStyles<typeof styles> & MenuCollapsibleItemP
...
+        }
+    };
+
+    const handlePopoverClose = (e: React.MouseEvent<HTMLElement>) => {
+        if (isMenuOpen) return;
+        const el = e.target as HTMLElement;
+        const rect = el.getBoundingClientRect();
+        const { clientX, clientY } = e;
+
+        if (childElements?.length && clientX + 2 > rect.right && clientY > rect.top && clientY < rect.bottom) return;
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:102]: What's the idea behind this?

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 5]
```diff
@@ -46,53 +31,160 @@ const CollapsibleItem: React.FC<WithStyles<typeof styles> & MenuCollapsibleItemP
...
     ...otherProps
 }) => {
+    OpenedIcon = OpenedIcon || (isMenuOpen ? ChevronUp : ChevronRight);
+    ClosedIcon = ClosedIcon || (isMenuOpen ? ChevronDown : ChevronRight);
+
     const itemLevel: MenuLevel["level"] = level ? level : 1;
     let hasSelectedChild = false;
     const location = useLocation();
 
+    const [isCollapsibleOpen, setIsCollapsibleOpen] = React.useState<boolean>(openByDefault || hasSelectedChild);
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:50]: This should have a better name. Maybe `isSubmenuOpen`?

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 6]
```diff
@@ -1,43 +1,28 @@
...
 
 export interface MenuCollapsibleItemProps extends MenuItemProps {
     children: MenuChild | MenuChild[];
     openByDefault?: boolean;
-    openedIcon?: React.ReactNode;
-    closedIcon?: React.ReactNode;
+    OpenedIcon?: React.JSXElementConstructor<SvgIconProps>;
+    ClosedIcon?: React.JSXElementConstructor<SvgIconProps>;
+    isMenuOpen?: boolean;
+    collapsedMenuTransitionComponent?: MenuProps["TransitionComponent"];
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:25]: Is there a need to override this?

File: packages/admin/admin/src/mui/menu/Item.styles.ts [hunk 7]
```diff
@@ -59,44 +75,82 @@ export const styles = (theme: Theme) =>
...
+            "& [class*='MuiListItemText-primary']": {
+                fontWeight: theme.typography.fontWeightRegular,
+                fontSize: 14,
+                lineHeight: "20px",
+                paddingLeft: ({ isMenuOpen }) => (isMenuOpen ? 14 : 0),
+                paddingTop: 14,
+                paddingBottom: 14,
+            },
+        },
+        level3Enumeration: {
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/Item.styles.ts:153]: What's the difference to `level3`?

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx [hunk 8]
```diff
@@ -2,51 +2,61 @@ import { Box, ComponentsOverrides, Theme, Tooltip, Typography } from "@mui/mater
...
-    if (drawerOpen === false) {
+    if (isMenuOpen === false) {
         title = getInitials(title);
     }
 
+    const childElements = React.Children.map(children, (child: MenuChild) => {
+        return React.cloneElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>(child, {
+            isMenuOpen,
+        });
+    });
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/ItemGroup.tsx:51]: With the menu context it shouldn't be necessary to clone the children

File: packages/admin/admin/src/mui/menu/Menu.tsx [hunk 9]
```diff
@@ -65,6 +66,12 @@ const MenuDrawer: React.FC<WithStyles<typeof styles> & MenuProps> = ({
     const temporaryOpen = variant === "temporary" && open;
     const permanentOpen = variant === "permanent" && open;
 
+    const childElements = React.Children.map(children, (child: React.ReactElement<MenuItemGroupProps>) => {
+        return React.cloneElement<MenuItemGroupProps>(child, {
+            isMenuOpen: open,
+        });
+    });
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/Menu.tsx:73]: Here as well

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 6 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx]: nah i removed it

File: packages/admin/admin/src/mui/menu/ItemGroup.tsx (hunk 8 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/ItemGroup.tsx]: Actually I can't really get rid of it because i need this prop for styling. It would create a big overhead.

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 2 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx]: You're right I changed it back

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 3 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:23]: Yeah it is but i need the prop for styling as stated below...

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 4 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:99]: This line checks if the mouse is over one of the child elements of the menu. If it is, the function returns and does nothing else. This is to prevent the menu from closing when the user is interacting with its contents.

File: packages/admin/admin/src/mui/menu/Item.styles.ts (hunk 7 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/Item.styles.ts:152]: It takes care of these lines before the element:

image

File: demo/admin/src/App.tsx (hunk 1 above)
Comment (jomunker) [reply demo/admin/src/App.tsx]: Done

File: packages/admin/admin/src/mui/menu/Item.tsx [hunk 10]
```diff
@@ -24,26 +25,28 @@ const Item: React.FC<WithStyles<typeof styles> & MenuItemProps & MuiListItemProp
     icon,
     level = 1,
     secondaryAction,
+    isMenuOpen,
     ...otherProps
 }) => {
     const context = React.useContext(MenuContext);
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/Item.tsx:31]: Why can't you use `context.open` from here?

Comment (jomunker) [reply packages/admin/admin/src/mui/menu/Item.tsx:31]: Because of conditional styling in the .styles file

**Comment by jomunker** [comment]
**Screen Rec of current styling:**
https://github.com/vivid-planet/comet/assets/56400587/b443779f-6d6d-4f39-a681-07ea5ebbf7bf

**Comment by johnnyomair** [comment]
I'll review this once #1234 has been merged. Setting as draft for now.

**Comment by johnnyomair** [comment]
@jomunker this has a lot of unrelated changes 🤔

**Comment by jomunker** [comment]
> @jomunker this has a lot of unrelated changes 🤔

Which ones?

**Comment by johnnyomair** [comment]
I was just looking at the 55 commits 🤔

**Comment by johnnyomair** [comment]
image

This looks wrong to me. Should there really be icons on second and third level?

**Comment by johnnyomair** [comment]
> The changes of the files itself look fine 🤔 maybe it would be better with a rebase but tbh don't wanna do one since it will take forever 😅

We'll leave this for now. For future stacking PRs please always rebase dependent PRs, and never merge the base PR. This will allow simpler rebases using `git rebase --onto <parent> HEAD~<number-of-commits-in-dependent-pr>`

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 3 above)
Comment (johnnyomair) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:25]: It should be possible to achieve the styling without cloning the children and adding this prop. Cloning the children is a legacy API not recommended by React: https://react.dev/reference/react/cloneElement

You could add a new class `menuOpen` that's conditionally added to the class list and sets the respective styles. I believe that's how you'd normally do this. @jamesricky what do you think?

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 11]
```diff
@@ -1,96 +1,192 @@
...
-            hasSelectedChild = true;
-        }
+    const [isSubmenuOpen, setIsSubmenuOpen] = React.useState<boolean>(openByDefault || hasSelectedChild.current);
+    const [anchorEl, setAnchorEl] = React.useState<HTMLElement | undefined>(undefined);
 
-        const newItemLevel = itemLevel + 1;
+    React.useEffect(() => {
+        // set open state manually to false to avoid a menu opening when isMenuOpen state changes
+        if (!isMenuOpen) setIsSubmenuOpen(false);
+    }, [isMenuOpen]);
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:55]: This should also be possible by doing

```tsx
// set open state manually to false to avoid a menu opening when isMenuOpen state changes
if (!isMenuOpen) {
  setIsSubmenuOpen(false);
}
```

without an effect. Docs: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

File: packages/admin/admin/src/mui/menu/Context.tsx [hunk 12]
```diff
@@ -3,17 +3,22 @@ import * as React from "react";
 export interface IMenuContext {
     open: boolean;
     toggleOpen: () => void;
+    drawerVariant: "temporary" | "permanent";
+    setDrawerVariant: React.Dispatch<React.SetStateAction<string>>;
```
Comment (johnnyomair) [inline packages/admin/admin/src/mui/menu/Context.tsx:7]: Why do we need this as part of the context?

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 3 above)
Comment (jamesricky) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:25]: Yes, if it's possible to add the class without `cloneElement` we should do that.

File: packages/admin/admin/src/mui/menu/Context.tsx (hunk 12 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/Context.tsx:7]: I added this to be able to access the drawerVariant. It is needed in the CollapsibleItem for rendering purposes. See [65cb263](https://github.com/vivid-planet/comet/pull/1240/commits/65cb26310aebd3a9e51ec57194565c079e384193). Without it the context menu rendered unintentionally in some cases.

File: demo/admin/src/common/MasterMenu.tsx [hunk 13]
```diff
@@ -24,10 +33,12 @@ const MasterMenu: React.FC = () => {
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [location]);
 
-    const sectionScopeTitle = `${scope?.domain.charAt(0).toUpperCase() + scope?.domain.slice(1)} - ${scope?.language?.toUpperCase()}`;
+    const sectionScopeTitle = `${scope?.domain.charAt(0).toUpperCase() + scope?.domain.slice(1)} - ${
+        values.language[0].label ?? scope?.language?.toUpperCase()
+    }`;
```
Comment (johnnyomair) [inline demo/admin/src/common/MasterMenu.tsx:38]: Why do you need to change this?

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 11 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx]: Just tested this again. Seems like it's needed. Otherwise there is an infinite render loop 😅

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 14]
```diff
@@ -1,96 +1,189 @@
...
-type MenuChild = React.ReactElement<MenuItemRouterLinkProps>;
+export type MenuChild = React.ReactElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>;
 
 export interface MenuCollapsibleItemProps extends MenuItemProps {
     children: MenuChild | MenuChild[];
     openByDefault?: boolean;
-    openedIcon?: React.ReactNode;
-    closedIcon?: React.ReactNode;
+    openedIcon?: React.JSXElementConstructor<SvgIconProps>;
+    closedIcon?: React.JSXElementConstructor<SvgIconProps>;
```
[the remark this answers is not in the document]
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:24]: You mean the type?

File: demo/admin/src/common/MasterMenu.tsx (hunk 13 above)
Comment (jomunker) [reply demo/admin/src/common/MasterMenu.tsx:38]: I actually don't know anymore 😅  But it was a dedicated commit (50a0794c5b34b28d63cde125a8bf16c0ab6714eb). Maybe @thomasdax98 can remember it.

File: demo/admin/src/common/MasterMenu.tsx [hunk 15]
```diff
@@ -2,19 +2,28 @@ import { Menu, MenuCollapsibleItem, MenuContext, MenuItemGroup, MenuItemRouterLi
...
     const useTemporaryMenu: boolean = windowSize.width < permanentMenuMinWidth;
 
+    useEffect(() => {
+        if (useTemporaryMenu) {
+            setDrawerVariant("temporary");
+        } else {
+            setDrawerVariant("permanent");
+        }
+    }, [setDrawerVariant, useTemporaryMenu]);
+
```
[the remark this answers is not in the document]
Comment (jomunker) [reply demo/admin/src/common/MasterMenu.tsx:26]: That depends I guess. We can move it into the menu component as default but use the prop if its passed. What do you think about that?

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 14 above)
[the remark this answers is not in the document]
Comment (johnnyomair) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:24]: Yes

File: demo/admin/src/common/MasterMenu.tsx (hunk 13 above)
Comment (johnnyomair) [reply demo/admin/src/common/MasterMenu.tsx:38]: BTW, the `values.language[0].label ?? scope?.language?.toUpperCase()` logic is incorrect since it always takes the first language, not the language of the current scope. I'd say we revert it.

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 14 above)
[the remark this answers is not in the document]
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:24]: It just makes it more restrictive to only allow icons for these props. I'll revert it.

File: demo/admin/src/common/MasterMenu.tsx (hunk 13 above)
Comment (thomasdax98) [reply demo/admin/src/common/MasterMenu.tsx:38]: The idea was displaying the label instead of the scope value ("English" instead of "EN")

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 14 above)
[the remark this answers is not in the document]
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:24]: image

Actually I need the icons to be callable as JSX and if we do it as `React.ReactNode` then TS has a problem calling the icons like in the screenshot.

File: demo/admin/src/common/MasterMenu.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (johnnyomair) [reply demo/admin/src/common/MasterMenu.tsx:26]: I'd say we move it into the component and update the context if the variant prop changes

**Comment by jamesricky** [comment]
@jomunker the implementation has some differences to the design:

- In the design both, the second and third level menus have a width of 240px, in the implementation there is no fixed width
- The paddings of both, the second and third level menus do not match the design (the menus, their items, and the item-title)
- The text in the second-level item can get too close to the arrow-icon, maybe make the text have a fixed width, as defined in the design (with overflow-ellipsis incase there is too much text)
- The font-styling of the item-title and the item-texts do not match the design 

Screenshot 2024-04-11 at 15 46 06

File: demo/admin/src/common/MasterMenu.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (jomunker) [reply demo/admin/src/common/MasterMenu.tsx:27]: Alright

**Comment by jomunker** [comment]
Thanks for checking it against the design 👍🏻  Do you think we should add a min width of 240px and a max width where the text is then truncated? One thing though: Wouldn't it look weird to have a fixed min width even if there are just one word items? That would be a lot of whitespace.

**Comment by jamesricky** [comment]
> Do you think we should add a min width of 240px and a max width where the text is then truncated?

I think the item should have a fixed width of 240px. 
The Text should be truncated when it exceeds it's available space, this can differ, depending on if the chevron-right icon is shown. 

> Wouldn't it look weird to have a fixed min width even if there are just one word items? That would be a lot of whitespace.

The fixed width was defined by the UX Team, I assume they took this into consideration. 
But they are always open to suggestions, if you want to follow up with them about this.

File: .changeset/eight-rules-laugh.md [hunk 16]
```diff
@@ -0,0 +1,7 @@
+---
+"@comet/admin": major
+---
+
+Change type of props `openedIcon` and `closedIcon` to `React.JSXElementConstructor<SvgIconProps>`
```
Comment (jamesricky) [inline .changeset/eight-rules-laugh.md:5]: Please mention the component of which the props are being changed

File: packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+        collapsibleIcon: {
+            fontSize: 12,
+            color: theme.palette.grey[200],
+        },
+        colorGrey: {
+            color: theme.palette.grey[200],
+        },
+        colorWhite: {
+            color: theme.palette.common.white,
+        },
```
Comment (jamesricky) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts:65]: The class-key should make it understandable, which element you are styling and in which state. 
Maybe they should be named something like `collapsibleIconLevelOne` and `collapsibleIconLevelTwo` rather than naming them after the styling they contain.

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 18]
```diff
@@ -1,96 +1,192 @@
...
+    const listClasses = [];
+    if (hasSelectedChild.current) listClasses.push(classes.childSelected);
+    if (isSubmenuOpen) listClasses.push(classes.open);
 
     return (
-        <div {...otherProps} className={listClasses.join(" ")}>
-            <div className={classes.listItem}>
+        <div {...otherProps}>
+            <div
+                id="menu-item"
```
Comment (jamesricky) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:117]: Ids should be unique per page, this one is being used for every item in the menu. 
Also `menu-item` is very generic, which increases the chance of colliding with another components id.

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 19]
```diff
@@ -1,96 +1,192 @@
...
+                    }
+                    className={listClasses.join(" ")}
                 />
             </div>
-            <Collapse in={open} timeout="auto" unmountOnExit>
-                <List disablePadding>{childElements}</List>
-            </Collapse>
+            {!isMenuOpen && drawerVariant === "permanent" ? (
+                <Menu
+                    id="mouse-over-menu"
```
Comment (jamesricky) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:154]: The id seems very generic, which increases the chance of colliding with another components id.

File: packages/admin/admin/src/mui/menu/Item.styles.ts [hunk 20]
```diff
@@ -3,7 +3,7 @@ import { createStyles } from "@mui/styles";
 
 import { MenuItemProps, MuiListItemProps } from "./Item";
 
-export type MenuItemClassKey = "root" | "level1" | "level2" | "level3" | "hasIcon" | "hasSecondaryText" | "hasSecondaryAction";
+export type MenuItemClassKey = "root" | "level1" | "level2" | "level3" | "hasIcon" | "hasSecondaryText" | "hasSecondaryAction" | "level3Enumeration";
```
Comment (jamesricky) [inline packages/admin/admin/src/mui/menu/Item.styles.ts:6]: I'm not sure the name really explains what this is key is used for... would something like `level3MenuOpen` be more fitting?

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 21]
```diff
@@ -1,96 +1,192 @@
...
+                        horizontal: "left",
+                    }}
+                    onClose={handlePopoverClose}
+                >
+                    <List disablePadding>
+                        {itemLevel === 1 && (
+                            <Typography className={classes.itemTitle} variant="h3">
+                                {primary}
+                            </Typography>
+                        )}
```
Comment (jamesricky) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:181]: Should the title be rendered before the list, instead of inside it, as it's not actually a list-item?

File: packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+    | "root"
+    | "childSelected"
+    | "listItem"
+    | "open"
+    | "collapsedMenuParentTitle"
+    | "itemTitle"
+    | "collapsibleIcon"
+    | "colorGrey"
+    | "colorWhite"
+    | "collapsibleOpen";
```
Comment (jamesricky) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts:16]: Is `collapsibleOpen` this actually being used?

File: packages/admin/admin/src/mui/menu/Menu.styles.ts [hunk 23]
```diff
@@ -57,7 +60,15 @@ export type MenuClassKey = "drawer" | "permanent" | "temporary" | "open" | "clos
...
 export const styles = () => {
     return createStyles<MenuClassKey, MenuProps>({
-        drawer: {},
+        drawer: {
+            "& > div": {
+                "&::-webkit-scrollbar": {
+                    display: "none",
+                },
+                scrollbarWidth: "none",
+                msOverflowStyle: "none",
```
[the remark this answers is not in the document]
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/Menu.styles.ts:69]: Because it changes the internal width of the objects and for something like a navbar i find it very weird and annoying if that happens.


https://github.com/vivid-planet/comet/assets/56400587/c7591d32-7f12-4905-9720-d36148728670

File: .changeset/eight-rules-laugh.md (hunk 16 above)
Comment (jomunker) [reply .changeset/eight-rules-laugh.md:5]: ✅

File: packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts (hunk 17 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts:65]: ✅

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 19 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:154]: ✅

File: packages/admin/admin/src/mui/menu/Item.styles.ts (hunk 20 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/Item.styles.ts:6]: ✅

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 21 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:181]: ✅

File: packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts (hunk 22 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts:16]: ✅

File: .changeset/eight-rules-laugh.md [hunk 24]
```diff
@@ -0,0 +1,7 @@
+---
+"@comet/admin": major
+---
+
+Change type of props `openedIcon` and `closedIcon` of `CollapsibleItem` to `React.JSXElementConstructor<SvgIconProps>`
```
Comment (jamesricky) [inline .changeset/eight-rules-laugh.md:5]: The component's exported name is `MenuCollapsibleItem`

```suggestion
Change type of props `openedIcon` and `closedIcon` of `MenuCollapsibleItem` to `React.JSXElementConstructor<SvgIconProps>`
```

File: demo/admin/src/common/MasterMenu.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (jomunker) [reply demo/admin/src/common/MasterMenu.tsx:27]: ✅

File: packages/admin/admin/src/mui/menu/Menu.styles.ts (hunk 23 above)
[the remark this answers is not in the document]
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/Menu.styles.ts]: ✅

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 25]
```diff
@@ -1,96 +1,201 @@
...
+        mouseOverMenuId = undefined;
+    if (typeof primary === "string") {
+        itemId = `menu-item-${primary?.replace(/\s/g, "-").toLowerCase()}`;
+        mouseOverMenuId = `item-hover-menu-${primary?.replace(/\s/g, "-").toLowerCase()}`;
+    }
 
     return (
-        <div {...otherProps} className={listClasses.join(" ")}>
-            <div className={classes.listItem}>
+        <div {...otherProps}>
```
Comment (jamesricky) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:121]: This is missing the `root` class-name. 
Should the `listClasses` not still be applied to this element, instead of the `MenuItem`?

File: packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts [hunk 26]
```diff
@@ -0,0 +1,60 @@
...
+import { createStyles } from "@mui/styles";
+
+import { MenuCollapsibleItemProps } from "./CollapsibleItem";
+
+export type MenuCollapsibleItemClassKey =
+    | "root"
+    | "childSelected"
+    | "listItem"
+    | "open"
+    | "collapsedMenuParentTitle"
```
Comment (jamesricky) [inline packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts:11]: Is this being used anywhere? If not, please remove the key and it's styling.

**Review by jamesricky — COMMENTED** [review]
Please compare your implementation to the design, this still doesn't look quite right (e.g. the item spacing and the submenu arrow)

Screenshot 2024-04-18 at 09 16 10

File: packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts (hunk 26 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts:11]: ✅

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx (hunk 25 above)
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx:121]: ✅

File: packages/admin/admin-theme/src/typographyOptions.ts
Comment (johnnyomair) [inline packages/admin/admin-theme/src/typographyOptions.ts:1]: The changes in this file seem somewhat unrelated to me... Shouldn't they be done in #1923?

Comment (jomunker) [reply packages/admin/admin-theme/src/typographyOptions.ts:1]: I needed the definition of subtitle2, overline and the font weight options for the menu to be correct in the future. I talked with @jamesricky and he said its fine if I add it here. Otherwise it would have been hard to implement without knowing how it will look like/ behave.

File: packages/admin/admin/src/mui/menu/CollapsibleItem.tsx [hunk 27]
```diff
@@ -1,96 +1,201 @@
...
+        if (anchorEl !== event.currentTarget) {
+            setAnchorEl(event.currentTarget);
+            setIsSubmenuOpen(true);
+        }
+    };
 
-    const [open, setOpen] = React.useState<boolean>(openByDefault || hasSelectedChild);
+    const handlePopoverClose = (e: React.MouseEvent<HTMLElement>) => {
+        if (isMenuOpen) return;
+        const el = e.target as HTMLElement;
```
[the remark this answers is not in the document]
Comment (jomunker) [reply packages/admin/admin/src/mui/menu/CollapsibleItem.tsx]: Good hint. Thanks! 👍🏻

**Review by jamesricky — APPROVED** [review]
